### DISCUSSION TITLE
How to show specific properties in many-to-one relation selectors like you can with many-to-many relations?

### DISCUSSION THREAD

**Discussion by gajdusek** [pimcore/pimcore#17522] [General]
Hello,

I'm working with many-to-one relations in Pimcore and want to display a custom field, specifically a **Calculated Value** property (e.g., `fullName` in my case), when selecting related objects. I noticed that in many-to-many relations, there's a **Visible Fields** option that allows you to select which fields are shown, making it easier to identify related objects. However, I couldn't find a similar option for many-to-one relations.

I also tried using the `Formatter service` to achieve this, but it doesn't seem to be applied when selecting objects in many-to-one relation fields (e.g., dropdowns).

Is there a way to enable **Visible Fields** for many-to-one relations or any other method to customize objects' representations in the relation selector?

Thanks in advance for any guidance!

Petr

**Comment by cancan101** [comment]
Ideally there would also be a "many-to-one _object_ relationship" which would be allowed only to objects and in turn allow choosing the "Visible fields".

The alternatives now are:
- Use many-to-many object relationship and set max items to `1`
- Use Select with an OptionProvider
- Use many-to-one with an optional Formatter

The Select for some use cases has the best UX including the ability for incremental search:
![image](https://github.com/user-attachments/assets/d6656c29-3d58-40a2-a579-8d8131d94385)
whereas that does not work for the many-to-many object (using display mode "tag field"):
![image](https://github.com/user-attachments/assets/a4f44f7b-0b4e-40e3-b610-7cfe87cdf85b)
